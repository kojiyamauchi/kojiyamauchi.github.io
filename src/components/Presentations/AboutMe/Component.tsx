/*
  About Me Component.
*/

import styles from './style.module.scss'

type Props = {
  closeAboutMe: () => void
}

export const AboutMe: React.FC<Props> = ({ closeAboutMe }): JSX.Element => {
  return (
    <div className={`${styles.wrapper} about-me`}>
      <h1 className={`${styles['primary-heading']} primary-heading`}>
        <span className="is-en">Koji Yamauchi</span>&nbsp;&nbsp;|&nbsp;&nbsp;<span className="is-jp">ヤマウチ コウジ</span>
      </h1>
      <p className={`${styles.letter} letter is-en`}>
        I am mainly involved in web front-end development and production from Based in Tokyo, Japan.
        <br />
        (Some back-end development and design work)
        <br />
        In the front-end area, we can handle everything from styling, such as UI implementation and design implementation, to interaction implementation,
        animation implementation, and API communication, so please feel free to contact us.
        <br />
        My skills, TypeScript / JavaScript / React / Next.js / Node.js / Webpack / Gulp / GitHub / Sass / CSS / HTML / VSCode / Figma(a little) / Sketch(a
        little) / Photoshop / Illustrator.
        <br />
        Currently, past projects is private.
        <br />
        If want you see please following contact us.
        <br />
        I’ll get back to you.
        <br />
        Loves &amp; fun making things.<span className="is-emoji">✌️</span>
      </p>
      <p className={`${styles.letter} letter is-jp`}>
        東京都内にて、主にWebフロントエンドの開発・制作に携わっております。
        <br />
        (Node.js開発・デザイン制作も少々)
        <br />
        フロントエンド領域においては、UI実装・デザイン実装等のスタイリング周りから、インタラクション実装・アニメーション実装・API通信まで承る事が可能ですので、お気軽にご連絡下さい。
        <br />
        使用言語・開発環境等は、TypeScript・JavaScript・React・Next.js・Node.js・Webpack・Gulp・GitHub・Sass・CSS・HTML・VSCode・Figma(少々)・Sketch(少々)・Photoshop・Illustratorになります。
        <br />
        現在、制作実績は非公開になりますので、下記よりお問い合わせ下さい。
        <br />
        折り返しご連絡させていただきます。
      </p>
      <div className={`${styles['mail-wrapper']} mail-wrapper`}>
        <a className={`${styles['button-mail']} button-mail`} href="mailto:all.the.small.things@me.com">
          <svg className={`${styles['icon-arrow-mail']} icon-arrow-mail`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z" />
          </svg>
          CONTACT US
        </a>
      </div>
      <div className={`${styles['back-wrapper']} back-wrapper`}>
        <button className={`${styles['button-back']} button-back`} onClick={closeAboutMe}>
          <svg className={`${styles['icon-arrow-back']} icon-arrow-back`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
          </svg>
          HIDE ABOUT ME
        </button>
      </div>
    </div>
  )
}
