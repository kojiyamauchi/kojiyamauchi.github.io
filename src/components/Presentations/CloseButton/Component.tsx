/*
  Close Button Component.
*/

import { useCloseButton } from './Hooks'
import styles from './style.module.scss'

type Props = {
  modalState: boolean
  modalClose: () => void
}

export const CloseButton: React.FC<Props> = ({ modalState, modalClose }): JSX.Element => {
  const { myComponent, letter } = useCloseButton(modalState)

  return (
    <button className={`${styles.wrapper} close-button`} ref={myComponent} onClick={(): void => modalClose()}>
      <span className="close-letters-wrapper">
        {Array.from(letter, (info, index) => (
          <span className={`${styles['close-letter']} close-letter`} key={index}>
            {info}
          </span>
        ))}
      </span>
    </button>
  )
}
