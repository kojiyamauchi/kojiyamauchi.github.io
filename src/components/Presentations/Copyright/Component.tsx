/*
  Copyright Component.
*/

import { useEffect, useState } from 'react'

import styles from './style.module.scss'

type Props = {
  className?: string
  parent: 'footer' | 'modal' | 'about-me' | 'error'
}

export const Copyright: React.VFC<Props> = ({ className, parent }): JSX.Element => {
  const [year, setYear] = useState<number | null>(null)

  useEffect(() => {
    setYear(new Date().getFullYear())
  }, [])

  return (
    <a className={`${styles.wrapper} ${className} is-${parent}`} href="http://kojiyamauchi.com" target="_blank" rel="noreferrer">
      Â© {year} All The Small Things, All Rights Reserved.
    </a>
  )
}
