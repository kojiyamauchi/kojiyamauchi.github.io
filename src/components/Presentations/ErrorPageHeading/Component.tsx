/*
  Error Page Heading Component.
*/
import styles from './style.module.scss'

export const ErrorPageHeading: React.VFC = (): JSX.Element => <h1 className={`${styles['primary-heading']} error-page-primary-heading`}>Error Page.</h1>
