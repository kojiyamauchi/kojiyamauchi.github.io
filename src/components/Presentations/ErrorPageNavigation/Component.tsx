/*
  Error Page Navigation Component.
*/

import Link from 'next/link'

import styles from './style.module.scss'

export const ErrorPageNavigation: React.FC = (): JSX.Element => {
  const link = [{ path: '/', letter: 'Back to Top Page', id: 1 }]

  const navigation = (): JSX.Element[] => {
    return link.map(
      (info): JSX.Element => (
        <li key={info.id} className="error-page-navi-list">
          <Link href={`${info.path}`}>
            <a className={`${styles.link} error-page-navi-link`}>{info.letter}</a>
          </Link>
        </li>
      )
    )
  }

  return <ul className={`${styles.wrapper} error-page-navigation`}>{navigation()}</ul>
}
