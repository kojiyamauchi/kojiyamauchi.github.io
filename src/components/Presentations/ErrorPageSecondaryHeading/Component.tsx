/*
  Error Pages Secondary Heading Component.
*/

import styles from './style.module.scss'

type Props = {
  letter: string
  translation?: boolean
}

export const ErrorPageSecondaryHeading: React.FC<Props> = ({ letter, translation }): JSX.Element => (
  <h2 className={`${styles['secondary-heading']} error-page-secondary-heading page-heading ${translation ? styles['is-translation'] : ''}`}>{letter}.ðŸ˜µ</h2>
)
