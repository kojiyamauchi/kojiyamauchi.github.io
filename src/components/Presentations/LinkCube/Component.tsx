/*
  Link Cube Component.
*/

import { DoubleSide } from 'three'

import { useLinkCube } from './Hooks'

export type Props = {
  textureName: string
  linkUrl: string
  positionValue: {
    x: {
      mobile: number
      tablet: number
      deskTop: number
    }
    y: {
      mobile: number
      tablet: number
      deskTop: number
    }
  }
  rotate: {
    x: number
    y: number
    z: number
  }
  storybook?: boolean
}

export const LinkCube: React.FC<Props> = ({ textureName, linkUrl, positionValue, rotate, storybook = false }): JSX.Element => {
  const { myComponent, memorizePositionX, memorizePositionY, memorizeCubeSize, texture } = useLinkCube(textureName, positionValue, rotate, storybook)

  return (
    <mesh
      ref={myComponent}
      position={[memorizePositionX, memorizePositionY, 0]}
      onClick={(): Window | null => open(linkUrl, '_blank')}
      onPointerOver={(): string => (document.body.style.cursor = 'pointer')}
      onPointerOut={(): string => (document.body.style.cursor = 'default')}
    >
      <boxGeometry attach="geometry" args={[memorizeCubeSize, memorizeCubeSize, memorizeCubeSize]} />
      <meshPhongMaterial attach="material" map={texture} side={DoubleSide} color="#ffffff" />
    </mesh>
  )
}
