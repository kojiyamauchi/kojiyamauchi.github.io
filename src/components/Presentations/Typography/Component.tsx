/*
  Typography Component.
*/

import styles from './style.module.scss'

const pathShadowArr = [
  'M359,222c0,0,12-68,57-172c0,0,3,101,43,172',
  'M366,148c0,0,43,8,61,5',
  'M496,50c0,0,9,73-2,177',
  'M546,50c0,0,12,44-4,177',
  'M646,71c0,0,65-4,95,5',
  'M696,68c0,0-18,30,7,148',
  'M769,44c0,0-2,2-3,17s-8,123-16,150',
  'M754,142c0,0,50-37,70,4c8,18,5,55,4,74',
  'M857,157c0,0,99,26,68-27c-20-31-62,5-61,21s14,95,74,41',
  'M1121,60c0,0-69-12-85,22s39,52,57,82s-11,42-30,55',
  'M1142,138c0,0,2,61,5,81s-2-65-2-65s12-24,34-25s34,74,34,74s-2-68,26-71c23-3,42,77,42,77',
  'M1385,143c0,0-43-23-60-2s-41,98,60,56c0,0,4-61,0-68c0,0,5,74,0,87',
  'M1431,49c0,0,10,104-3,176',
  'M1481,49c0,0,7,37-4,176',
  'M1581,71c0,0,53-6,95,5',
  'M1631,68c0,0-17,62,6,148',
  'M1704,42c0,0-4,7-4,19s-4,98-16,149',
  'M1689,145c0,0,51-46,70,12c3,14,5,40,3,62',
  'M1809,99c0,0-6,1-3,5',
  'M1809,135v27c0,0,3,34-4,60',
  'M1861,131c0,0-1,58-9,91c0,0-4-124,54-67c6,10,5,77,5,77',
  'M1990,126c0,0-36-16-45,63c0,21,71,40,64,4s-22-81-22-81s87,232-51,176',
  'M2114,131c0,0-52-22-70,24c-9,23,83,2,78,35s-49,36-79,11',
  'M2159,209c0,0,4,14-23,34',
  'M75.2,50.2c-8.7,33.5-10.7,68.3-12.3,102.9c-1.2,24.9-2.1,50,1.6,74.6',
  'M78.2,48.4c12.1,2.1,23.9,6.1,34.9,11.7c6,3.1,11.9,6.7,16.4,11.7c4.6,5,7.7,11.5,7.7,18.3c-0.1,5.9-2.5,11.4-5.3,16.6c-10.1,18.8-25,34.9-43,46.3c-5.8,3.7-12.4,7-19.1,5.8c-7.8-1.3-13.3-7.9-18.1-14.2c11.7,23.2,33.6,39.3,55.2,53.7c9.4,6.3,19.1,12.5,30,15.6',
  'M165.2,156.9c17,2,34.2,3,51.3,3c7.3,0,15.6-0.6,20.3-6.1c3.4-4,4.1-10,2.4-15s-5.5-9.2-9.9-12.1c-8.2-5.5-18.8-7.2-28.3-4.5c-9.5,2.7-17.6,9.8-21.7,18.8c-5.8,12.9-3.1,27.9,1.3,41.4c1.7,5.4,3.8,10.9,7.8,15c2.1,2.2,4.6,3.9,7.3,5.4c13,7,29.4,7.3,42.7,0.8c4.7-2.3,9.1-5.6,11.3-10.3',
  'M302.6,145.3c0.2,1.9,0.1,3.9-0.3,5.7',
  'M298.7,215.5c0.7,1.7,0.7,3.6,0.2,5.3'
]

const pathTypographyArr = [
  'M349,212c0,0,12-68,57-172c0,0,3,101,43,172',
  'M356,138c0,0,43,8,61,5',
  'M486,40c0,0,9,73-2,177',
  'M536,40c0,0,12,44-4,177',
  'M636,61c0,0,65-4,95,5',
  'M686,58c0,0-18,30,7,148',
  'M759,34c0,0-2,2-3,17s-8,123-16,150',
  'M744,132c0,0,50-37,70,4c8,18,5,55,4,74',
  'M847,147c0,0,99,26,68-27c-20-31-62,5-61,21s14,95,74,41',
  'M1111,50c0,0-69-12-85,22s39,52,57,82s-11,42-30,55',
  'M1132,128c0,0,2,61,5,81s-2-65-2-65s12-24,34-25s34,74,34,74s-2-68,26-71c23-3,42,77,42,77',
  'M1375,133c0,0-43-23-60-2s-41,98,60,56c0,0,4-61,0-68c0,0,5,74,0,87',
  'M1421,39c0,0,10,104-3,176',
  'M1471,39c0,0,7,37-4,176',
  'M1571,61c0,0,53-6,95,5',
  'M1621,58c0,0-17,62,6,148',
  'M1694,32c0,0-4,7-4,19s-4,98-16,149',
  'M1679,135c0,0,51-46,70,12c3,14,5,40,3,62',
  'M1799,89c0,0-6,1-3,5',
  'M1799,125v27c0,0,3,34-4,60',
  'M1851,121c0,0-1,58-9,91c0,0-4-124,54-67c6,10,5,77,5,77',
  'M1980,116c0,0-36-16-45,63c0,21,71,40,64,4s-22-81-22-81s87,232-51,176',
  'M2104,121c0,0-52-22-70,24c-9,23,83,2,78,35s-49,36-79,11',
  'M2149,199c0,0,4,14-23,34',
  'M65.2,40.2c-8.7,33.5-10.7,68.3-12.3,102.9c-1.2,24.9-2.1,50,1.6,74.6',
  'M68.2,38.4c12.1,2.1,23.9,6.1,34.9,11.7c6,3.1,11.9,6.7,16.4,11.7c4.6,5,7.7,11.5,7.7,18.3c-0.1,5.9-2.5,11.4-5.3,16.6c-10.1,18.8-25,34.9-43,46.3c-5.8,3.7-12.4,7-19.1,5.8c-7.8-1.3-13.3-7.9-18.1-14.2c11.7,23.2,33.6,39.3,55.2,53.7c9.4,6.3,19.1,12.5,30,15.6',
  'M155.2,146.9c17,2,34.2,3,51.3,3c7.3,0,15.6-0.6,20.3-6.1c3.4-4,4.1-10,2.4-15c-1.7-5-5.5-9.2-9.9-12.1c-8.2-5.5-18.8-7.2-28.3-4.5c-9.5,2.7-17.6,9.8-21.7,18.8c-5.8,12.9-3.1,27.9,1.3,41.4c1.7,5.4,3.8,10.9,7.8,15c2.1,2.2,4.6,3.9,7.3,5.4c13,7,29.4,7.3,42.7,0.8c4.7-2.3,9.1-5.6,11.3-10.3',
  'M292.6,135.3c0.2,1.9,0.1,3.9-0.3,5.7',
  'M288.7,205.5c0.7,1.7,0.7,3.6,0.2,5.3'
]

export type Props = {
  visited: boolean
}

export const Typography: React.VFC<Props> = ({ visited }): JSX.Element => {
  return (
    <div className={styles.wrapper} id="typography-component">
      <svg
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
        x="0px"
        y="0px"
        viewBox="0 0 2188.2 325.7"
        enableBackground="new 0 0 2188.2 325.7"
        xmlSpace="preserve"
      >
        <g className="shadows">
          {pathShadowArr.map((info, index) => (
            <path className={`draw-shadow ${styles['draw-shadow']} path-${index + 1} ${visited ? '' : 'is-introduction'}`} d={info} key={index + 1} />
          ))}
        </g>
        <g className="typographys">
          {pathTypographyArr.map((info, index) => (
            <path className={`draw-typography ${styles['draw-typography']} path-${index + 1} ${visited ? '' : 'is-introduction'}`} d={info} key={index + 1} />
          ))}
        </g>
      </svg>
      <h1 className={`${styles['primary-heading']} primary-heading`}>
        Koji Yamauchi | ヤマウチ コウジ | 山内 康次 | Tokyo | Japan | Freelance Web Developer | Web Frontend | TypeScript | JavaScript | React | Next.js |
        Webpack | Gulp | Node.js | Design | Photoshop | Illustrator
      </h1>
    </div>
  )
}
