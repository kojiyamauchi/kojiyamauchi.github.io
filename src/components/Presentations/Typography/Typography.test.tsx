/*
  Typography Component Unit Test.
*/

import { Typography } from './Component'
import { render /* ,screen */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
// import userEvent from '@testing-library/user-event'

describe('Typography Component Unit Test', () => {
  it('Snap Shot Testing', () => {
    const { asFragment } = render(<Typography visited={true} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
