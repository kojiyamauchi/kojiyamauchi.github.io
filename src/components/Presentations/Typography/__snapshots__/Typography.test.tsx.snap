// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component Unit Test Snap Shot Testing 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
    id="typography-component"
  >
    <svg
      enable-background="new 0 0 2188.2 325.7"
      version="1.1"
      viewBox="0 0 2188.2 325.7"
      x="0px"
      xml:space="preserve"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
      y="0px"
    >
      <g
        class="shadows"
      >
        <path
          class="draw-shadow draw-shadow path-1 "
          d="M359,222c0,0,12-68,57-172c0,0,3,101,43,172"
        />
        <path
          class="draw-shadow draw-shadow path-2 "
          d="M366,148c0,0,43,8,61,5"
        />
        <path
          class="draw-shadow draw-shadow path-3 "
          d="M496,50c0,0,9,73-2,177"
        />
        <path
          class="draw-shadow draw-shadow path-4 "
          d="M546,50c0,0,12,44-4,177"
        />
        <path
          class="draw-shadow draw-shadow path-5 "
          d="M646,71c0,0,65-4,95,5"
        />
        <path
          class="draw-shadow draw-shadow path-6 "
          d="M696,68c0,0-18,30,7,148"
        />
        <path
          class="draw-shadow draw-shadow path-7 "
          d="M769,44c0,0-2,2-3,17s-8,123-16,150"
        />
        <path
          class="draw-shadow draw-shadow path-8 "
          d="M754,142c0,0,50-37,70,4c8,18,5,55,4,74"
        />
        <path
          class="draw-shadow draw-shadow path-9 "
          d="M857,157c0,0,99,26,68-27c-20-31-62,5-61,21s14,95,74,41"
        />
        <path
          class="draw-shadow draw-shadow path-10 "
          d="M1121,60c0,0-69-12-85,22s39,52,57,82s-11,42-30,55"
        />
        <path
          class="draw-shadow draw-shadow path-11 "
          d="M1142,138c0,0,2,61,5,81s-2-65-2-65s12-24,34-25s34,74,34,74s-2-68,26-71c23-3,42,77,42,77"
        />
        <path
          class="draw-shadow draw-shadow path-12 "
          d="M1385,143c0,0-43-23-60-2s-41,98,60,56c0,0,4-61,0-68c0,0,5,74,0,87"
        />
        <path
          class="draw-shadow draw-shadow path-13 "
          d="M1431,49c0,0,10,104-3,176"
        />
        <path
          class="draw-shadow draw-shadow path-14 "
          d="M1481,49c0,0,7,37-4,176"
        />
        <path
          class="draw-shadow draw-shadow path-15 "
          d="M1581,71c0,0,53-6,95,5"
        />
        <path
          class="draw-shadow draw-shadow path-16 "
          d="M1631,68c0,0-17,62,6,148"
        />
        <path
          class="draw-shadow draw-shadow path-17 "
          d="M1704,42c0,0-4,7-4,19s-4,98-16,149"
        />
        <path
          class="draw-shadow draw-shadow path-18 "
          d="M1689,145c0,0,51-46,70,12c3,14,5,40,3,62"
        />
        <path
          class="draw-shadow draw-shadow path-19 "
          d="M1809,99c0,0-6,1-3,5"
        />
        <path
          class="draw-shadow draw-shadow path-20 "
          d="M1809,135v27c0,0,3,34-4,60"
        />
        <path
          class="draw-shadow draw-shadow path-21 "
          d="M1861,131c0,0-1,58-9,91c0,0-4-124,54-67c6,10,5,77,5,77"
        />
        <path
          class="draw-shadow draw-shadow path-22 "
          d="M1990,126c0,0-36-16-45,63c0,21,71,40,64,4s-22-81-22-81s87,232-51,176"
        />
        <path
          class="draw-shadow draw-shadow path-23 "
          d="M2114,131c0,0-52-22-70,24c-9,23,83,2,78,35s-49,36-79,11"
        />
        <path
          class="draw-shadow draw-shadow path-24 "
          d="M2159,209c0,0,4,14-23,34"
        />
        <path
          class="draw-shadow draw-shadow path-25 "
          d="M75.2,50.2c-8.7,33.5-10.7,68.3-12.3,102.9c-1.2,24.9-2.1,50,1.6,74.6"
        />
        <path
          class="draw-shadow draw-shadow path-26 "
          d="M78.2,48.4c12.1,2.1,23.9,6.1,34.9,11.7c6,3.1,11.9,6.7,16.4,11.7c4.6,5,7.7,11.5,7.7,18.3c-0.1,5.9-2.5,11.4-5.3,16.6c-10.1,18.8-25,34.9-43,46.3c-5.8,3.7-12.4,7-19.1,5.8c-7.8-1.3-13.3-7.9-18.1-14.2c11.7,23.2,33.6,39.3,55.2,53.7c9.4,6.3,19.1,12.5,30,15.6"
        />
        <path
          class="draw-shadow draw-shadow path-27 "
          d="M165.2,156.9c17,2,34.2,3,51.3,3c7.3,0,15.6-0.6,20.3-6.1c3.4-4,4.1-10,2.4-15s-5.5-9.2-9.9-12.1c-8.2-5.5-18.8-7.2-28.3-4.5c-9.5,2.7-17.6,9.8-21.7,18.8c-5.8,12.9-3.1,27.9,1.3,41.4c1.7,5.4,3.8,10.9,7.8,15c2.1,2.2,4.6,3.9,7.3,5.4c13,7,29.4,7.3,42.7,0.8c4.7-2.3,9.1-5.6,11.3-10.3"
        />
        <path
          class="draw-shadow draw-shadow path-28 "
          d="M302.6,145.3c0.2,1.9,0.1,3.9-0.3,5.7"
        />
        <path
          class="draw-shadow draw-shadow path-29 "
          d="M298.7,215.5c0.7,1.7,0.7,3.6,0.2,5.3"
        />
      </g>
      <g
        class="typographys"
      >
        <path
          class="draw-typography draw-typography path-1 "
          d="M349,212c0,0,12-68,57-172c0,0,3,101,43,172"
        />
        <path
          class="draw-typography draw-typography path-2 "
          d="M356,138c0,0,43,8,61,5"
        />
        <path
          class="draw-typography draw-typography path-3 "
          d="M486,40c0,0,9,73-2,177"
        />
        <path
          class="draw-typography draw-typography path-4 "
          d="M536,40c0,0,12,44-4,177"
        />
        <path
          class="draw-typography draw-typography path-5 "
          d="M636,61c0,0,65-4,95,5"
        />
        <path
          class="draw-typography draw-typography path-6 "
          d="M686,58c0,0-18,30,7,148"
        />
        <path
          class="draw-typography draw-typography path-7 "
          d="M759,34c0,0-2,2-3,17s-8,123-16,150"
        />
        <path
          class="draw-typography draw-typography path-8 "
          d="M744,132c0,0,50-37,70,4c8,18,5,55,4,74"
        />
        <path
          class="draw-typography draw-typography path-9 "
          d="M847,147c0,0,99,26,68-27c-20-31-62,5-61,21s14,95,74,41"
        />
        <path
          class="draw-typography draw-typography path-10 "
          d="M1111,50c0,0-69-12-85,22s39,52,57,82s-11,42-30,55"
        />
        <path
          class="draw-typography draw-typography path-11 "
          d="M1132,128c0,0,2,61,5,81s-2-65-2-65s12-24,34-25s34,74,34,74s-2-68,26-71c23-3,42,77,42,77"
        />
        <path
          class="draw-typography draw-typography path-12 "
          d="M1375,133c0,0-43-23-60-2s-41,98,60,56c0,0,4-61,0-68c0,0,5,74,0,87"
        />
        <path
          class="draw-typography draw-typography path-13 "
          d="M1421,39c0,0,10,104-3,176"
        />
        <path
          class="draw-typography draw-typography path-14 "
          d="M1471,39c0,0,7,37-4,176"
        />
        <path
          class="draw-typography draw-typography path-15 "
          d="M1571,61c0,0,53-6,95,5"
        />
        <path
          class="draw-typography draw-typography path-16 "
          d="M1621,58c0,0-17,62,6,148"
        />
        <path
          class="draw-typography draw-typography path-17 "
          d="M1694,32c0,0-4,7-4,19s-4,98-16,149"
        />
        <path
          class="draw-typography draw-typography path-18 "
          d="M1679,135c0,0,51-46,70,12c3,14,5,40,3,62"
        />
        <path
          class="draw-typography draw-typography path-19 "
          d="M1799,89c0,0-6,1-3,5"
        />
        <path
          class="draw-typography draw-typography path-20 "
          d="M1799,125v27c0,0,3,34-4,60"
        />
        <path
          class="draw-typography draw-typography path-21 "
          d="M1851,121c0,0-1,58-9,91c0,0-4-124,54-67c6,10,5,77,5,77"
        />
        <path
          class="draw-typography draw-typography path-22 "
          d="M1980,116c0,0-36-16-45,63c0,21,71,40,64,4s-22-81-22-81s87,232-51,176"
        />
        <path
          class="draw-typography draw-typography path-23 "
          d="M2104,121c0,0-52-22-70,24c-9,23,83,2,78,35s-49,36-79,11"
        />
        <path
          class="draw-typography draw-typography path-24 "
          d="M2149,199c0,0,4,14-23,34"
        />
        <path
          class="draw-typography draw-typography path-25 "
          d="M65.2,40.2c-8.7,33.5-10.7,68.3-12.3,102.9c-1.2,24.9-2.1,50,1.6,74.6"
        />
        <path
          class="draw-typography draw-typography path-26 "
          d="M68.2,38.4c12.1,2.1,23.9,6.1,34.9,11.7c6,3.1,11.9,6.7,16.4,11.7c4.6,5,7.7,11.5,7.7,18.3c-0.1,5.9-2.5,11.4-5.3,16.6c-10.1,18.8-25,34.9-43,46.3c-5.8,3.7-12.4,7-19.1,5.8c-7.8-1.3-13.3-7.9-18.1-14.2c11.7,23.2,33.6,39.3,55.2,53.7c9.4,6.3,19.1,12.5,30,15.6"
        />
        <path
          class="draw-typography draw-typography path-27 "
          d="M155.2,146.9c17,2,34.2,3,51.3,3c7.3,0,15.6-0.6,20.3-6.1c3.4-4,4.1-10,2.4-15c-1.7-5-5.5-9.2-9.9-12.1c-8.2-5.5-18.8-7.2-28.3-4.5c-9.5,2.7-17.6,9.8-21.7,18.8c-5.8,12.9-3.1,27.9,1.3,41.4c1.7,5.4,3.8,10.9,7.8,15c2.1,2.2,4.6,3.9,7.3,5.4c13,7,29.4,7.3,42.7,0.8c4.7-2.3,9.1-5.6,11.3-10.3"
        />
        <path
          class="draw-typography draw-typography path-28 "
          d="M292.6,135.3c0.2,1.9,0.1,3.9-0.3,5.7"
        />
        <path
          class="draw-typography draw-typography path-29 "
          d="M288.7,205.5c0.7,1.7,0.7,3.6,0.2,5.3"
        />
      </g>
    </svg>
    <h1
      class="primary-heading primary-heading"
    >
      Koji Yamauchi | ヤマウチ コウジ | 山内 康次 | Tokyo | Japan | Freelance Web Developer | Web Frontend | TypeScript | JavaScript | React | Next.js | Webpack | Gulp | Node.js | Design | Photoshop | Illustrator
    </h1>
  </div>
</DocumentFragment>
`;
